crates/hir_ty/src/diagnostics/decl_check.rs

```diff
@@ -869,10 +901,36 @@ fn main() {
             r#"
             #[allow(non_snake_case)]
     fn NonSnakeCaseName(SOME_VAR: u8) -> u8{
+        // cov_flags generated output from elsewhere in this file
+        extern "C" {
+            #[no_mangle]
+            static lower_case: u8;
+        }
+
         let OtherVar = SOME_VAR + 1;
         OtherVar
     }
 
+    #[allow(nonstandard_style)]
+    mod CheckNonstandardStyle {
+        fn HiImABadFnName() {}
+    }
+
+    #[allow(bad_style)]
+    mod CheckBadStyle {
+        fn HiImABadFnName() {}
+    }
+
+    mod F {
+        #![allow(non_snake_case)]
+        fn CheckItWorksWithModAttr(BAD_NAME_HI: u8) {}
+    }
+
+    trait BAD_TRAIT {
+        fn BAD_FUNCTION();
+        fn BadFunction();
+    }
```

Comment:
Why don't we report anything here? Was this preexisting?

Reply:
As far as I know, yes. It's because the trait isn't visited in the pass that picks up the items, which I haven't changed. I put it there more to document the current behaviour but I probably should move it to make it clear it's not intentional and should not be preserved.

Reply:
Yeah, if you could add a FIXME comment that'd be great